frontend/model/contracts/identity.js

```diff
@@ -31,16 +64,12 @@ sbp('chelonia/defineContract', {
     'gi.contracts/identity': {
       validate: (data, { state }) => {
         objectMaybeOf({
-          attributes: objectMaybeOf({
-            username: string,
-            email: string,
-            picture: unionOf(string, objectOf({
-              manifestCid: string,
-              downloadParams: optional(object)
-            }))
-          })
+          attributes: attributesType
         })(data)
         const { username } = data.attributes
+        if (!username) {
+          throw new TypeError('A username is required')
+        }
```

Comment:
Below this line are the same checks that are now in `validateUsername` - please DRY